### PR TITLE
Groups in DocTags

### LINKED ISSUE #7336: Specify group of the component as doctag or section.
To make defining component groups easier, the entry's group should be specified in its docstring. This is a first step of implementing https://github.com/orgs/enso-org/discussions/7120.

The group list should be still in the package.yaml, and the "export" fields there may be honored.

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Fixes #7336 in a quick way.

Next to the old way of defining groups, the library can just add `GROUP` tag to some entities, and it will be added to the group specified in tag's description.

The group name may be qualified (with project name, like `Standard.Base.Input/Output`) or just name - in the latter case, IDE will assume a group defined in the same library as the entity.

Also moved some entities from "export" list in package.yaml to GROUP tag to give an example. I didn't move all of those, as I assume the library team will reorganize those groups anyway.

### Important Notes

@jdunkerley @radeusgd @GregoryTravis When you will start specifying groups in tags, remember that:
* The groups still belongs to a concrete project; if some entity outside a project wants to be added to its group, the "qualified" name should be specified. See `Table.new` example in this PR.
* If the group name does not reflect any group in package.yaml **the tag is ignored**.
* A single entity may be only in a single group. If it's specified in both package.yaml and in tag, the tag takes precedence.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/src/controller/searcher.rs [hunk 1]
```diff
@@ -39,6 +40,8 @@ pub mod input;
 /// needed. Currently enabled to trigger engine's caching of user-added nodes.
 /// See: https://github.com/enso-org/ide/issues/1067
 pub const ASSIGN_NAMES_FOR_NODES: bool = true;
+/// A name of component group containin entries representing literals.
```
Comment (vitvakatu) [inline app/gui/src/controller/searcher.rs]:
```suggestion
/// A name of a component group containing entries representing literals.
```

File: app/gui/src/controller/searcher.rs [hunk 2]
```diff
@@ -717,9 +720,9 @@ impl Searcher {
 
 fn add_virtual_entries_to_builder(builder: &mut component::Builder) {
     let snippets = component::hardcoded::INPUT_SNIPPETS.with(|s| s.clone());
-    let group_name = component::hardcoded::INPUT_GROUP_NAME;
-    let project = project::QualifiedName::standard_base_library();
-    builder.add_virtual_entries_to_group(group_name, project, snippets);
+    // Unwrap is safe because conversion from INPUt_GROUP_NAME is tested.
```
Comment (vitvakatu) [inline app/gui/src/controller/searcher.rs]:
```suggestion
    // Unwrap is safe because conversion from INPUT_GROUP_NAME is tested.
```

**Review by vitvakatu — APPROVED** [review]

**Review by JaroslavTulach — APPROVED** [review]
Does this solution also renders further work #6497 unnecessary? Oh, my procrastination and laziness...

**Comment by jdunkerley** [comment]
I believe the longer term approach is to go for attributes still.

**Comment by vitvakatu** [comment]
Yep, our plans is to use doc tags instead of attributes before the support for attributes lands.

**Comment by kazcw** [comment]
# QA: :green_circle:

**Comment by farmaazon** [comment]
@jdunkerley @radeusgd @GregoryTravis I need an accept from one of you, as I changed libraries' code.

File: distribution/lib/Standard/Base/0.0.0-dev/package.yaml [hunk 3]
```diff
@@ -10,16 +10,8 @@ maintainers:
...
-          - Standard.Base.System.File.File.new
-          - Standard.Base.Data.read
-          - Standard.Base.Data.read_text
-          - Standard.Base.Data.list_directory
-    - Web:
-        exports:
-          - Standard.Base.Network.HTTP.HTTP.new
-          - Standard.Base.Data.fetch
+    - Input: {}
+    - Web: {}
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/package.yaml:14]: Do the groups still need to be defined in `package.yaml`?

What if we annotate a method with a `GROUP xyz` when `xyz` is not defined in `package.yaml`?

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/package.yaml:14]: Yes the groups have to be defined in the package.yaml.
It is supposed to generate an error but that may not be done yet,

**Review by jdunkerley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/package.yaml (hunk 3 above)
Comment (farmaazon) [reply distribution/lib/Standard/Base/0.0.0-dev/package.yaml:14]: IDE will print the error log and the entry will not be assigned to any group.

**Pull request merged**